src/test/java/com/diana/restaurant/controllers/finalProduct/FinalProductControllerTest.java

```diff
@@ -0,0 +1,83 @@
+package com.diana.restaurant.controllers.finalProduct;
+
+import com.diana.restaurant.controllers.FinalProductController;
+import com.diana.restaurant.entities.FinalProduct;
+import com.diana.restaurant.services.implementation.FinalProductServiceImpl;
+import com.diana.restaurant.services.interfaces.FinalProductService;
+import com.diana.restaurant.util.fixtures.services.FinalProductServiceFixtures;
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+public class FinalProductControllerTest {
+  private FinalProductController finalProductController;
+  private FinalProductService finalProductServiceMock;
+
+  @BeforeEach
+  public void setup() {
+    FinalProductService finalProductService = new FinalProductServiceImpl();
+    finalProductServiceMock = Mockito.mock(finalProductService.getClass());
+    finalProductController = new FinalProductController(finalProductServiceMock);
+  }
+
+  @Test
+  public void getAll_ShouldReturnListOfFinalProducts() {
+    var finalProducts = FinalProductServiceFixtures.buildFinalProducts(3);
+    Mockito.when(finalProductServiceMock.findAll()).thenReturn(finalProducts);
+    Assertions.assertEquals(finalProducts, finalProductController.getAll());
+    Mockito.verify(finalProductServiceMock, Mockito.times(1)).findAll();
+  }
+
+  @Test
+  public void add_ShouldInsertFinalProduct() {
+    var finalProduct = FinalProductServiceFixtures.buildFinalProduct();
+    finalProduct.setId(1L);
+    Mockito.when(finalProductServiceMock.insert(finalProduct)).thenReturn(finalProduct);
+    Assertions.assertEquals(finalProduct, finalProductController.add(finalProduct));
+    var finalProductToTestVerify = FinalProductServiceFixtures.buildFinalProduct(finalProduct);
+    Mockito.verify(finalProductServiceMock, Mockito.times(1))
```

Comment:
Why are you inserting an entity with an id? Does that even make sense?

Reply:
Changed it to:

Reply:
you need to set an id, since that is not the current behavior of the service

Reply:
solved